pupil_src/shared_modules/imu_timeline.py

```diff
@@ -0,0 +1,718 @@
+                    self.q[0] * self.q[0]
+                    - self.q[1] * self.q[1]
+                    - self.q[2] * self.q[2]
+                    + self.q[3] * self.q[3],
+                )
+            )
+            + 90
+        )
+
+
+class IMURecording:
+    DTYPE_RAW = np.dtype(
+        [
+            ("gyro_x", "<f4"),
+            ("gyro_y", "<f4"),
+            ("gyro_z", "<f4"),
+            ("accel_x", "<f4"),
+            ("accel_y", "<f4"),
+            ("accel_z", "<f4"),
+        ]
+    )
+
+    def __init__(self, path_to_imu_raw: pathlib.Path):
+        stem = path_to_imu_raw.stem
+        self.path_raw = path_to_imu_raw
+        self.path_ts = path_to_imu_raw.with_name(stem + "_timestamps.npy")
+        self.load()
+
+    def load(self):
+        if not self.path_raw.exists() and self.path_ts.exists():
+            self.ts = np.empty(0, dtype=np.float64)
+            self.raw = np.empty(0, dtype=self.DTYPE_RAW)
+            return
+
+        self.ts = np.load(str(self.path_ts))
+        self.raw = np.fromfile(str(self.path_raw), dtype=self.DTYPE_RAW).view(
+            np.recarray
+        )
+        num_ts_during_init = self.ts.size - len(self.raw)
+        if num_ts_during_init > 0:
+            self.ts = self.ts[num_ts_during_init:]
+
+
+class IMUTimeline(Plugin):
+    """
+    plot and export imu data
+    export: imu_timeline.csv
+    keys:
+        imu_timestamp: timestamp of the source image frame
+        world_index: associated_frame: closest world video frame
+        gyro_x: angular velocity about the x axis in degrees/s
+        gyro_y: angular velocity about the y axis in degrees/s
+        gyro_z: angular velocity about the z axis in degrees/s
+        accel_x: linear acceleration along the x axis in G (9.80665 m/s^2)
+        accel_y: linear acceleration along the y axis in G (9.80665 m/s^2)
+        accel_z: linear acceleration along the z axis in G (9.80665 m/s^2)
+        pitch: orientation expressed as Euler angles
+        roll: orientation expressed as Euler angles
+    See Pupil docs for relevant coordinate systems
+    """
+
+    gyro_error = 50
+    should_draw_raw = False
+    should_draw_orientation = False
+
+    IMU_PATTERN_RAW = r"^extimu ps(\d+).raw"
+
+    CMAP = {
+        "gyro_x": cygl_utils.RGBA(0.12156, 0.46666, 0.70588, 1.0),
+        "gyro_y": cygl_utils.RGBA(1.0, 0.49803, 0.05490, 1.0),
+        "gyro_z": cygl_utils.RGBA(0.17254, 0.62745, 0.1725, 1.0),
+        "accel_x": cygl_utils.RGBA(0.83921, 0.15294, 0.15686, 1.0),
+        "accel_y": cygl_utils.RGBA(0.58039, 0.40392, 0.74117, 1.0),
+        "accel_z": cygl_utils.RGBA(0.54901, 0.33725, 0.29411, 1.0),
+        "pitch": cygl_utils.RGBA(0.12156, 0.46666, 0.70588, 1.0),
+        "roll": cygl_utils.RGBA(1.0, 0.49803, 0.05490, 1.0),
+    }
+    NUMBER_SAMPLES_TIMELINE = 4000
+    TIMELINE_LINE_HEIGHT = 16
+    icon_chr = chr(0xE922)
+    icon_font = "pupil_icons"
```

Comment:
This currently corresponds to the `Vis Polyline` plugin icon. I suggest finding an alternative icon.